File: gluecodium/src/test/resources/smoke/field_constructors/output/android/com/example/smoke/FieldConstructorsAllDefaults.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
public final class FieldConstructorsAllDefaults {
    @NonNull
    public String stringField;
    public int intField;
    public boolean boolField;
    public FieldConstructorsAllDefaults() {
        this.stringField = "nonsense";
        this.intField = 42;
        this.boolField = true;
    }
    public FieldConstructorsAllDefaults(final int intField) {
        this.intField = intField;
        this.stringField = "nonsense";
        this.boolField = true;
    }
    public FieldConstructorsAllDefaults(final int intField, @NonNull final String stringField) {
        this.intField = intField;
        this.stringField = stringField;
        this.boolField = true;
    }
    public FieldConstructorsAllDefaults(final boolean boolField, final int intField, @NonNull final String stringField) {
        this.boolField = boolField;
        this.intField = intField;
        this.stringField = stringField;
    }
}

File: gluecodium/src/test/resources/smoke/field_constructors/output/android-kotlin/com/example/smoke/FieldConstructorsAllDefaults.kt
/*

 *
 */

@file:JvmName("FieldConstructorsAllDefaultsExtensions")

package com.example.smoke


class FieldConstructorsAllDefaults {
    @JvmField var stringField: String
    @JvmField var intField: Int
    @JvmField var boolField: Boolean





    constructor() {
        this.stringField = "nonsense"
        this.intField = 42
        this.boolField = true
    }


    constructor(intField: Int) {
        this.intField = intField
        this.stringField = "nonsense"
        this.boolField = true
    }


    constructor(intField: Int, stringField: String) {
        this.intField = intField
        this.stringField = stringField
        this.boolField = true
    }


    constructor(boolField: Boolean, intField: Int, stringField: String) {
        this.boolField = boolField
        this.intField = intField
        this.stringField = stringField
    }





}

